### PR TITLE
Fix input value not being set if the value was '0'

### PR DESCRIPTION
**Pull request by patrickpatrickpatrick**
## What

Input component now uses the govukAttributes macro to validate and format attributes. Test and example added for edge-case outlined in issue https://github.com/alphagov/govuk-frontend/issues/4669.

## Why

Issue https://github.com/alphagov/govuk-frontend/issues/4669 outlines a case where setting 'value' of the Input component to 0 (integer) would not set the value of the input component. This is because the template logic would see '0' as false-y and so not set the value attribute. The govukAttributes takes this case into account and so using it means that this no longer occurs. 

This is based from https://github.com/alphagov/govuk-frontend/pull/4770. It adds tests and the input wrapper classes and attributes back.

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/components/input/template.njk [hunk 1]
```diff
@@ -3,25 +3,64 @@
...
+      disabled: {
+        value: params.disabled,
+        optional: true
+      },
+      "aria-describedby": {
+        value: describedBy,
+        optional: true
+      },
+      autocomplete: {
+        value: params.autocomplete | default("none", true)
```
Comment (36degrees) [inline packages/govuk-frontend/src/govuk/components/input/template.njk:52]: It looks like this has changed from #4770 and changes the behaviour slightly.

Previously, if `params.autocomplete` wasn't set, we wouldn't output an `autocomplete` attribute.

Now, if `params.autocomplete` isn't set, we output a default `autocomplete="none"` [which isn't valid](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete#values).

This is one of the reasons the tests are failing – we run an html validator against all of the examples and [it's erroring with '"none" is not a valid autocomplete token or field name'](https://github.com/alphagov/govuk-frontend/actions/runs/8922309070/job/24504321827?pr=4963#step:6:483).

File: packages/govuk-frontend/src/govuk/components/input/template.njk [hunk 2]
```diff
@@ -3,25 +3,64 @@
...
 {%- macro _inputElement(params) -%}
-  <input class="govuk-input {%- if params.classes %} {{ params.classes }}{% endif %} {%- if params.errorMessage %} govuk-input--error{% endif %}" id="{{ params.id }}" name="{{ params.name }}" type="{{ params.type | default("text", true) }}"
-    {%- if (params.spellcheck === false) or (params.spellcheck === true) %} spellcheck="{{ params.spellcheck }}"{% endif %}
-    {%- if params.value %} value="{{ params.value }}"{% endif %}
-    {%- if params.disabled %} disabled{% endif %}
-    {%- if describedBy %} aria-describedby="{{ describedBy }}"{% endif %}
-    {%- if params.autocomplete %} autocomplete="{{ params.autocomplete }}"{% endif %}
-    {%- if params.pattern %} pattern="{{ params.pattern }}"{% endif %}
-    {%- if params.inputmode %} inputmode="{{ params.inputmode }}"{% endif %}
-    {%- if params.autocapitalize %} autocapitalize="{{ params.autocapitalize }}"{% endif %}
```
Comment (36degrees) [inline packages/govuk-frontend/src/govuk/components/input/template.njk:24]: **issue** It looks like the `autocapitalize` attribute was added in #4442, after the work in #4770 was done, so this will need adding to the new `govukAttributes` call.

(There's a corresponding test for this which we need to keep too)

File: packages/govuk-frontend/src/govuk/components/input/template.test.js [hunk 3]
```diff
@@ -455,22 +446,4 @@ describe('Input', () => {
       expect($prefixBeforeSuffix.length).toBeTruthy()
     })
   })
-
-  describe('when it includes the input wrapper', () => {
```
Comment (36degrees) [inline packages/govuk-frontend/src/govuk/components/input/template.test.js:459]: **issue** Why have these tests been removed?

File: packages/govuk-frontend/src/govuk/components/input/input.yaml [hunk 4]
```diff
@@ -399,6 +399,14 @@ examples:
...
         text: With value
       value: QQ 12 34 56 C
+  - name: zero value
+    hidden: true
+    options:
+      id: with-zero-value
+      name: with-zero-value
+      label:
+        text: With zero value
+      value: !!int 0
```
Comment (36degrees) [inline packages/govuk-frontend/src/govuk/components/input/input.yaml:409]: **question** Is the casting definitely necessary here? My understanding was that YAML should treat this is an integer already.

File: packages/govuk-frontend/src/govuk/components/input/template.test.js [hunk 5]
```diff
@@ -71,6 +71,13 @@ describe('Input', () => {
       expect($component.val()).toBe('QQ 12 34 56 C')
     })
 
+    it('renders with zero value', () => {
```
Comment (36degrees) [inline packages/govuk-frontend/src/govuk/components/input/template.test.js:74]: **praise** Thanks for adding a test for this 🙌🏻

**Review by 36degrees — COMMENTED** [review]
Thanks for picking this up and congrats on your first contribution to GOV.UK Frontend on the team 😎 🎉 

There's a couple of things that aren't quite right at the minute – in particular it looks like the changes in #4770 might have been out of date with some other changes to add the `autocapitalize` attribute, sorry we'd missed that!

Let me know if it'd be helpful to chat through any of this or if you've got any questions!

**Review by 36degrees — APPROVED** [review]
Looks good to me! 🙌🏻
